pkg/hvs/tasks/create_default_template.go

```diff
@@ -0,0 +1,460 @@
+/*
+ * Copyright (C) 2020 Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+package tasks
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/intel-secl/intel-secl/v3/pkg/hvs/constants"
+	"github.com/intel-secl/intel-secl/v3/pkg/model/hvs"
+
+	"github.com/intel-secl/intel-secl/v3/pkg/hvs/postgres"
+	commConfig "github.com/intel-secl/intel-secl/v3/pkg/lib/common/config"
+	"github.com/intel-secl/intel-secl/v3/pkg/lib/common/setup"
+	"github.com/pkg/errors"
+)
+
+type CreateDefaultTemplate struct {
+	DBConf  commConfig.DBConfig
+	deleted []string
+
+	commandName   string
+	TemplateStore *postgres.FlavorTemplateStore
+}
+
+var defaultFlavorTemplateNames = []string{
+	"default-linux-tpm20-tboot",
+	"default-linux-tpm20-suefi",
+	"default-linux-tpm20-cbnt",
+	"default-uefi",
+	"default-bmc",
+	"default-pfr",
+}
+
+func (t *CreateDefaultTemplate) Run() error {
+	var templates []hvs.FlavorTemplate
+
+	ftStore, err := t.flavorTemplateStore()
+	if err != nil {
+		return errors.Wrap(err, "Failed to initialize flavor template store instance")
+	}
+
+	if len(t.deleted) != 0 {
+		// Recover deleted default template.
+		err := ftStore.Recover(t.deleted)
+		if err != nil {
+			return errors.Wrapf(err, "Failed to recover default flavor template(s) %s", t.deleted)
+		}
+		t.deleted = []string{}
+		return nil
+	}
+
+	templates, err = getTemplates()
+	if err != nil {
+		return err
+	}
+
+	for _, ft := range templates {
+		_, err := ftStore.Create(&ft)
+		if err != nil {
+			return errors.Wrap(err, "Failed to create default flavor template with ID \""+ft.ID.String()+"\"")
+		}
+	}
+
+	return nil
+}
+
+func (t *CreateDefaultTemplate) Validate() error {
+	ftStore, err := t.flavorTemplateStore()
+	if err != nil {
+		return errors.Wrap(err, "Failed to initialize flavor template store instance")
+	}
+	var ftList []hvs.FlavorTemplate
+	defaultFlavorTemplateMap := map[string]bool{}
+	t.deleted = []string{}
+
+	for _, templateName := range defaultFlavorTemplateNames {
+		defaultFlavorTemplateMap[templateName] = false
+	}
+
+	ftList, err = ftStore.Search(false)
+	if err != nil {
+		return errors.Wrap(err, "Failed to validate "+t.commandName)
+	}
+	if len(ftList) == 0 {
+		return errors.New("No active templates found in db")
+	}
+
+	for _, template := range ftList {
+		defaultFlavorTemplateMap[template.Label] = true
+	}
+
+	for _, templateName := range defaultFlavorTemplateNames {
+		if !defaultFlavorTemplateMap[templateName] {
+			t.deleted = append(t.deleted, templateName)
+		}
+	}
+
+	if len(t.deleted) != 0 {
```

Comment:
Templates are lazy deleted, in table it will be marked as true, during next installation true->false alone can be done, if that entry itself not there in the table (manual removal) then that template alone added newly this is to avoid duplicates. Or need to modify this?

Reply:
if there is no restriction on default template deletion then this should be fine